### ISSUE TITLE
[bug] Task hasher looks at VCS root rather than workspace root

### ISSUE THREAD

**Issue by dvicory** [moonrepo/moon#1388]
**Describe the bug**

**Steps to reproduce**

1. Create a workspace within a subdirectory of the VCS root, e.g. `frontend`.
2. Setup packages, apps, etc.
3. Define a basic task, e.g.
```
$schema: 'https://moonrepo.dev/schemas/tasks.json'

tasks:
  test-prettier:
    command: 'prettier --check .'
```
4. Within `frontend`, run `moon run :test-prettier`

**Expected behavior**

I expect the task to run, without warnings about trying to find files from the VCS root, rather than from the workspace root.

**Screenshots**

```
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/catalog-info.yaml but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/primitives/Button.tsx but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/primitives/index.ts but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/styles/unistyles.ts but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/primitives/Button.stories.tsx but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/package.json but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/primitives/types.ts but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/styles/breakpoints.ts but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/styles/index.ts but it does not exist, skipping
[ WARN 16:49:31.099] moon_task_hasher::task_hasher  Attempted to hash input frontend/packages/design-system/index.ts but it does not exist, skipping
```

This repeats on for all the files within each application or library defined.

The directory structure looks something like this (slightly redacted, but structured the same):

```
. (VCS root)
├── CODEOWNERS
├── README.md
├── backend (Rails app)
└── frontend (moon workspace root)
    ├── apps
    │   ├── design-system-storybook
    │   │   ├── moon.yml
    │   │   ├── package.json
    │   └── resident
    │       ├── moon.yml
    │       └── package.json
    ├── package.json
    ├── packages
    │   └── design-system
    │       ├── moon.yml
    │       └── package.json
    └── yarn.lock
```

**Environment**

```
  System:
    OS: macOS 13.6.4
    CPU: (10) arm64 Apple M1 Max
    Memory: 49.66 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.1 - ~/.asdf/installs/nodejs/18.19.1/bin/node
    Yarn: 4.1.1 - ~/.asdf/installs/yarn/1.22.10/bin/yarn
    npm: 10.2.4 - ~/.asdf/installs/nodejs/18.19.1/bin/npm
  Managers:
    Cargo: 1.67.0 - /opt/homebrew/bin/cargo
    CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    Homebrew: 4.2.12 - /opt/homebrew/bin/brew
    pip3: 22.0.4 - ~/.asdf/installs/python/3.10.4/bin/pip3
    RubyGems: 3.5.3 - ~/.asdf/installs/ruby/3.3.0/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 15.0.0 - /usr/bin/gcc
    Git: 2.39.2 - /opt/homebrew/bin/git
    Clang: 15.0.0 - /usr/bin/clang
    Curl: 8.4.0 - /usr/bin/curl
  Servers:
    Apache: 2.4.56 - /usr/sbin/apachectl
  Virtualization:
    Docker: 24.0.1 - /opt/homebrew/bin/docker
  SDKs:
    iOS SDK:
      Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, watchOS 10.2
  IDEs:
    Android Studio: 2023.1 AI-231.9392.1.2311.11330709
    Atom: 1.58.0
    Vim: 9.0 - /usr/bin/vim
    Xcode: 15.1/15C65 - /usr/bin/xcodebuild
  Languages:
    Bash: 3.2.57 - /bin/bash
    Go: 1.18.2 - /opt/homebrew/bin/go
    Java: 19.0.2 - /usr/bin/javac
    Perl: 5.30.3 - /usr/bin/perl
    Python: 3.10.4 - ~/.asdf/installs/python/3.10.4/bin/python
    Python3: 3.10.4 - ~/.asdf/installs/python/3.10.4/bin/python3
    Ruby: 3.3.0 - ~/.asdf/installs/ruby/3.3.0/bin/ruby
    Rust: 1.67.0 - /opt/homebrew/bin/rustc
  Databases:
    SQLite: 3.39.5 - /usr/bin/sqlite3
  Browsers:
    Chrome: 122.0.6261.129
    Chrome Canary: 124.0.6366.2
    Safari: 17.3
    Safari Technology Preview: 17.4
  Monorepos:
    Yarn Workspaces: 4.1.1
```

**Additional context**

I did some sleuthing and did see previous fixes for VCS root vs workspace root, so it seems this may be a regression, or at least one for task hasher. This is my first experience with moonrepo, so it's possible I'm missing something.

**Comment by milesj** [comment]
@dvicory What version of moon are you on? Having trouble reproducing this on the latest.

**Comment by milesj** [comment]
Ok fixed in 1.22.8

**Comment by dvicory** [comment]
Thanks @milesj, 1.22.8 fixed our use case!
